htdocs/societe/class/societe.class.php

```diff
@@ -3904,12 +3904,21 @@ public function create_from_member(Adherent $member, $socname = '', $socalias =
 		global $conf, $user, $langs;
 
 		dol_syslog(get_class($this)."::create_from_member", LOG_DEBUG);
+		$fullname = $member->getFullName($langs);
 
-		$name = $socname ? $socname : $member->societe;
-		if (empty($name)) {
-			$name = $member->getFullName($langs);
+		if ($member->morphy == 'mor') {
+			$socname = $member->company? $member->company : $member->societe;
```

Comment:
If $socname or $socalias was forced by caller, here you erased the value.
I suggest to implement the logic to define socname and socalias in this method only if input var are not forced by caller, for compatibility and to keep the use of existing parameters.
So you should add a test` if (empty($socname))`